### PR TITLE
Simplify `requirements.txt`, update actions workflow

### PR DESCRIPTION
**Pull request by ewels**
The combination of a fairly old Python setup step and a _huge_ `requirements.txt` file is breaking the build for the new automation over in `nf-tower-cloud` repo.

I think we're probably better off keeping `requirements.txt` down to a minimum, it'll make it much easier for people to run, though I'm not sure what the history is here. Do we really need to install Jupyter to build the docs?

Also including some other fixes that I needed to build locally, plus a couple of minor things I noticed.

### CODE REVIEW

File: overrides/main.html [hunk 1]
```diff
@@ -1,7 +1,5 @@
 {% extends "partials/announce.html" %}
 
-{% include "announce.html" %}
-
```
Comment (ewels) [inline overrides/main.html:4]: We're extending from `announce.html` already in the line above, so I don't think that we need to include it as well.

Also, the file path is wrong (should be `partials/announce.html`). The build failed for me with this here.

File: overrides/partials/base.html [hunk 2]
```diff
@@ -1,11 +1 @@
...
-
-{% block analytics %}
-<script async src="https://www.googletagmanager.com/gtag/js?id=UA-364526-12"></script>
-<script>
-  window.dataLayer = window.dataLayer || [];
-  function gtag(){dataLayer.push(arguments);}
-  gtag('js', new Date());
-  gtag('config', 'UA-364526-12');
-</script>
-{% endblock %}
```
Comment (ewels) [inline overrides/partials/base.html:11]: We have the google analytics ID set in `mkdocs.yml` already, so I don't think that we need to include it a second time here.

Also, this is the old universal analytics ID, not the new GA-4 property. This one will stop working in a few days.

**Comment by ewels** [comment]
Docs seem to be building with this nicely: https://github.com/seqeralabs/nf-tower-cloud/pull/5245

Installation is quite a bit faster - 53 seconds instead of 2-3 minutes.

File: overrides/main.html (hunk 1 above)
Comment (llewellyn-sl) [reply overrides/main.html:4]: Thanks. I'd raised a revert PR to remove that line already, but I can close that PR if we're removing it here and build succeeds. Will build locally now before approving.

**Review by llewellyn-sl — APPROVED** [review]
Local build works, and the announce banner is showing. Thanks Phil! Approved.

**Pull request merged**